system/security/LdapSecurity/ldapsecurity.cpp

```diff
@@ -1234,10 +1234,11 @@ IAuthMap * CLdapSecManager::createAuthMap(IPropertyTree * authconfig)
                 location->getProp("@required", required);
                 location->getProp("@description", description);
                 
-                if(pathstr.length() == 0)
-                    throw MakeStringException(-1, "path empty in Authenticate/Location");
                 if(rstr.length() == 0)
                     throw MakeStringException(-1, "resource empty in Authenticate/Location");
+                if(pathstr.length() == 0)
+                    throw MakeStringException(-1, "path empty in Authenticate/Location for service '%s'", rstr.str());
```

Comment:
do we know that rstr is not empty at this point? 
Also, is rstr "@resource" always going to map to a service?

Reply:
We do know rstr is not empty because of the 2 previous lines.
I reworded the message to state "resource" instead of "service"
Please rereview